### PR TITLE
[mtls-proxy] Add annotations option

### PR DESCRIPTION
**Pull request by gracedo**
https://jira.d2iq.com/browse/D2IQ-64099
Adds the option to add annotations to the proxy deployment (will be used to add reloader annotation for secrets `secret.reloader.stakater.com/reload`)

When this is merged, the following needs to be done:
1. Bump the version of `mtls-proxy` used in `kommander-thanos` and `kommander-karma` and add the `secret.reloader.stakater.com/reload` annotations
1. Bump the deps of `kommander-thanos` and `kommander-karma` in `kommander` chart to pull in the latest changes from the above
1. Bump kommander version in `kubeaddons-kommander`

## Testing
I tested this manually by hosting `mtls-proxy`, `kommander-thanos`, `kommander-karma`, and `kommander charts`. I `konvoy up` on addons version `master`, then upgraded the kommander chart by deploying addons with my `kubeaddons-kommander` branch `gracedo/bug_certs_D2IQ-64099`.
```
  - configRepository: https://github.com/mesosphere/kubeaddons-kommander
    #configVersion: master
    configVersion: gracedo/bug_certs_D2IQ-64099
    addonsList:
    - name: kommander
      enabled: true
```

Upon upgrading, I verified that updated secrets were federated out to the managed clusters (as they were before) then checked that the proxies now had the reloader annotation:
image
image

I checked Thanos and Karma UIs in the kommander cluster and saw that they were still able to connect to the proxies even after upgrading
image
image

### CODE REVIEW

File: stable/mtls-proxy/Chart.yaml [hunk 1]
```diff
@@ -3,6 +3,6 @@ appVersion: "1.0"
 description: mTLS proxy
 name: mtls-proxy
 home: https://github.com/mesosphere/charts
-version: 0.1.0
+version: 0.1.1
```
Comment (branden) [inline stable/mtls-proxy/Chart.yaml]: Baby chart's first bugfix 😭

Comment (gracedo) [reply stable/mtls-proxy/Chart.yaml]: 🚼graduating to toddler!

File: stable/mtls-proxy/templates/deployment.yaml [hunk 2]
```diff
@@ -4,6 +4,10 @@ metadata:
   name: {{ include "mtls-proxy.fullname" . }}
   labels:
 {{ include "mtls-proxy.labels" . | indent 4 }}
+{{- with .Values.deployment.annotations }}
```
Comment (alejandroEsc) [inline stable/mtls-proxy/templates/deployment.yaml]: just to double check, can you verify that empty annotation here behaves and deploys as expected.

Comment (alejandroEsc) [reply stable/mtls-proxy/templates/deployment.yaml]: ok tested this, looks great, thank!

**Pull request merged**